### PR TITLE
fix: Correct the use of GetContacts() #5012

### LINKED ISSUE #5012: Email alerts not working with AD authentication
DO NOT DELETE THIS INFORMATION.

> Please read this information carefully.

GitHub issues is for feature requests or bugs, please do not post issues asking for help or how to do X, Y or Z. 
You can use our irc channel ##librenms on freenode to ask questions or our [community site](https://community.librenms.org).

- [x] Is your install up to date? [Updating your install](http://docs.librenms.org/General/Updating/)
      Please do not submit an issue if your install is not up to date within the last 24 hours or on a stable monthly release.
- [x] Please include all of the information between the `====================================` section of `./validate.php`.
==========================================================
Component | Version
--------- | -------
LibreNMS  | c46de37197b7b7db70f50c5ef7a3f00c423789f4
DB Schema | 150
PHP       | 5.5.9-1ubuntu4.20
MySQL     | 5.5.49-0ubuntu0.14.04.1
RRDTool   | 1.4.7
SNMP      | NET-SNMP 5.7.2
==========================================================

- [ ] If you would like us to add a new device then please provide the information asked for [here](http://docs.librenms.org/Support/FAQ/#faq20)
- [x] Please provide as much detail as possible.

Email alerts stops working when AD authentication is enabled. Eventlog gives the following error: Could not issue critical alert for rule 'Service up/down' to transport 'mail' Error: You must provide at least one recipient email address. This error is encountered on both my Librenms test boxes running Ubuntu 14.04 and 16.04.

Doing a test mail transport, it is able to send the test mail to all users in the AD group. Even with only one user being a member of the AD group, no alerts are sent.

The setup is the following: A "Networking" AD group contains three user accounts each with assigned email addresses.
In config.php, the following is added: $config['auth_ad_groups']['Networking']['level'] = 10;

Alerting settings: https://drive.google.com/open?id=0B8WCN9DjlJUca1pXajQxYXNGMlk
Alert capture: https://drive.google.com/open?id=0B8WCN9DjlJUcYkw4X3JMaUZFek0
Alert rules: https://drive.google.com/open?id=0B8WCN9DjlJUcNENsSGd6VWdDYTg

### PR DESCRIPTION
**Pull request by laf**
DO NOT DELETE THIS TEXT

#### Please note

> Please read this information carefully. You can run `./scripts/pre-commit.php` to check your code before submitting.

- [x] Have you signed the [Contributors agreement](http://docs.librenms.org/General/Contributing/) - please do NOT submit a pull request unless you have (signing the agreement in the same pull request is fine). Your commit message for signing the agreement must appear as per the docs.
- [x] Have you followed our [code guidelines?](http://docs.librenms.org/Developing/Code-Guidelines/)

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`

Possibility that this fixes #5012

### CODE REVIEW

File: html/includes/output/query.inc.php [hunk 1]
```diff
@@ -43,7 +43,7 @@
                 $rule['query'] = GenSQL($rule['rule']);
             }
             $sql = $rule['query'];
-            $qry = dbFetchRow($sql, array($device_id));
+            $qry[] = dbFetchRow($sql, array($device_id));
```
Comment (murrant) [inline html/includes/output/query.inc.php]: Won't this work better `$qry = dbFetchRows($sql, array($device_id));`?
`Row`->`Rows`

File: html/includes/output/query.inc.php [hunk 2]
```diff
@@ -43,7 +43,7 @@
...
             if (is_array($qry)) {
```
Comment (murrant) [inline html/includes/output/query.inc.php]: This is now always true.

File: html/includes/output/query.inc.php (hunk 1 above)
Comment (laf) [inline html/includes/output/query.inc.php]: No, $qry is passed later on and needs to be an array, the $sql is a single row lookup but we're in a loop so need to build that $qry array up.

File: html/includes/output/query.inc.php (hunk 2 above)
Comment (laf) [inline html/includes/output/query.inc.php]: Updated

**Pull request merged**